### PR TITLE
More Detailed Installation & Configuration

### PR DESCRIPTION
**Pull request by moriarty**
Added sections to Installation
- Recommended
- Alternative

Added Configuration section

Added to the FAQ and reorganized (to easily reference the most common patch)
- I've had several questions via messages and email: They almost always require the same patch.

### CODE REVIEW

**Comment by moriarty** [comment]
@svenschneider 

I did these edits using the GitHub in browser editor, so the GitHub Flavoured Markdown renders properly.

These instructions recommend 14.04, Boost 1.54. Because they almost work out of the box. 
A patch is still required. 

The patch is actually almost always required. Perhaps it's best to apply the Makefile Patch in the rockin branch, and include a .patch file to undo the patch for the rare cases were the patch causes problems.

**Comment by moriarty** [comment]
@svenschneider

File: README.md [hunk 1]
```diff
@@ -5,30 +5,74 @@ This is the repository of the referee box for the RoCKIn competition (http://roc
...
+The RoCKIn Central Factory Hub (CFH) can be installed on most linux distributions. However, some require dependencies being built from source. The current recommended setup is Ubuntu 14.04, with Boost 1.54. We support and test on Ubuntu 14.04 LTS. For other non-supported distributions and setups see Alternative Installation Hints at the end of this readme.
 
-Additionally, the following debian packages need to be installed:
+### Officially Supported Setup: Ubuntu 14.04, Boost 1.54
 
-    sudo apt-get install libssl-dev libelf-dev mongodb-dev mongodb-clients mongodb libzmq3-dev
-  
-## Compiling the RoCKIn@Work Referee Box
-### Cloning the Git repository:
+1. The CFH is based off LLSF refbox, and uses [clips](http://clipsrules.sourceforge.net/). Add [Tim's PPA](https://launchpad.net/~timn/+archive/ubuntu/clips) for clips:
```
Comment (svenschneider) [inline README.md]: 1. LLSF is already mentioned in the beginning, no need to repeat here.
2. No need to explicitly mention the use of CLIPS (else which dependencies to enumerate, which not?)
3. Tim -> Tim Niemueller

File: README.md [hunk 2]
```diff
@@ -5,30 +5,74 @@ This is the repository of the referee box for the RoCKIn competition (http://roc
 
 
 ## Installation
-To install the RoCKIn referee box, please install the prerequisites of the LLSF referee box which are described here: https://trac.fawkesrobotics.org/wiki/LLSFRefBox/Install
+The RoCKIn Central Factory Hub (CFH) can be installed on most linux distributions. However, some require dependencies being built from source. The current recommended setup is Ubuntu 14.04, with Boost 1.54. We support and test on Ubuntu 14.04 LTS. For other non-supported distributions and setups see Alternative Installation Hints at the end of this readme.
```
Comment (svenschneider) [inline README.md]: 1. linux -> Linux
2. "The current recommended setup is Ubuntu 14.04, with Boost 1.54. We support and test on Ubuntu 14.04 LTS." -> "The currently supported setup is Ubuntu 14.04, with Boost 1.54".

File: README.md [hunk 3]
```diff
@@ -5,30 +5,74 @@ This is the repository of the referee box for the RoCKIn competition (http://roc
...
+### Officially Supported Setup: Ubuntu 14.04, Boost 1.54
 
-    sudo apt-get install libssl-dev libelf-dev mongodb-dev mongodb-clients mongodb libzmq3-dev
-  
-## Compiling the RoCKIn@Work Referee Box
-### Cloning the Git repository:
+1. The CFH is based off LLSF refbox, and uses [clips](http://clipsrules.sourceforge.net/). Add [Tim's PPA](https://launchpad.net/~timn/+archive/ubuntu/clips) for clips:
+      
+        sudo add-apt-repository ppa:timn/clips
+    (Note: This PPA is maintained by the developers of the LLSF referee box. And only works for Ubuntu 12.04, 12.10 and 14.04. For others, see Alternative Setup.)
```
Comment (svenschneider) [inline README.md]: -> "(Note: This PPA currently only works for Ubuntu 12.04, 12.10 and 14.04.)"
Other setups are already mentioned in the beginning.

File: README.md [hunk 4]
```diff
@@ -5,30 +5,74 @@ This is the repository of the referee box for the RoCKIn competition (http://roc
...
+    
+2. Update apt:
+        
+        sudo apt-get update
```
Comment (svenschneider) [inline README.md:19]: Merge into the next section, i.e.:

```
sudo apt-get update
sudo apt-get install ...
```

File: README.md [hunk 5]
```diff
@@ -5,30 +5,74 @@ This is the repository of the referee box for the RoCKIn competition (http://roc
...
+3. Install the dependencies for both LLSFRB and CFH
+        
+        sudo apt-get install libmodbus-dev libclips-dev clips libclipsmm-dev \
+                             protobuf-compiler libprotobuf-dev libprotoc-dev \
+                             boost1.54-all-dev libmodbus-dev \
+                             libglibmm-2.4-dev libgtkmm-3.0-dev \
+                             libncursesw5-dev libyaml-cpp-dev libavahi-client-dev git \
+                             libssl-dev libelf-dev mongodb-dev mongodb-clients \
+                             mongodb libzmq3-dev
+    (Note: Boost 1.54 is specified to avoid causing apt-get broken package problems with ROS. If you are using another version of Boost see Alternative Setup.)
```
Comment (svenschneider) [inline README.md]: -> "(Note: Force Boost 1.54 to avoid conflicts with certain packages)"?

File: README.md [hunk 6]
```diff
@@ -5,30 +5,74 @@ This is the repository of the referee box for the RoCKIn competition (http://roc
...
 
+4. Clone the CFH repo
```
Comment (svenschneider) [inline README.md]: "Clone this repository"

File: README.md [hunk 7]
```diff
@@ -5,30 +5,74 @@ This is the repository of the referee box for the RoCKIn competition (http://roc
...
+        
+        git clone https://github.com/rockin-robot-challenge/at_work_central_factory_hub.git
 
-### Compiling the refbox:
+5. Enter the CFH repo
```
Comment (svenschneider) [inline README.md]: "Build the Central Factory Hub:

```
cd at_work_central_factory_hub
make
```

"

File: README.md [hunk 8]
```diff
@@ -5,30 +5,74 @@ This is the repository of the referee box for the RoCKIn competition (http://roc
...
+
+1. Configure the CFH with your editor
     
+        editor ./cfg/config.yaml
+
+    For testing the following changes are common:
+    - Change the CFH IP address in section "shell", this should be where you're running the CFH. Other machines communicating with the CFH need this value set correctly.
+        
+            refbox-host: !ipv4 localhost
+    - disable mongodb, all messages will be logged and in early developement this may not be desired:
```
Comment (svenschneider) [inline README.md]: Well, disabling MongoDB results in no messages being logged (not all messages being logged).

File: README.md [hunk 9]
```diff
@@ -5,30 +5,74 @@ This is the repository of the referee box for the RoCKIn competition (http://roc
...
+        
+            enable: false
+    - If testing on the same machine as the CFH is running, toggle the comments on "port" or "send-port" and "recv-port". To run the rockin-viewer and rockin-controller on the same machine, change in the section "public-peer", to run the robot on the same machine, the TeamName-peer secion must be edited.
```
Comment (svenschneider) [inline README.md]: 1. "If the robot software or CFH clients (e.g. GUIs) are running on the same machine as the CFH (e.g. local testing), ..."
2. For CFH clients the "path" of the configuration properties is "llsfrb/comm/public-peer".
3. For CFH peers (robots) the "path" of the configuration properties is "llsfrb/comm/<team-name>-peer".
4. "toggle the comments on "port" or "send-port" and "recv-port"" not sure if this so clear?!
5. "secion" -> "section"

File: README.md (hunk 1 above)
Comment (moriarty) [inline README.md]: I only mentioned CLIPS because it's the only dependency which requires a PPA. 
I'll remove it

File: README.md [hunk 10]
```diff
@@ -43,28 +87,39 @@ How can I solve the problem?
...
+### Alternative Installation Hints
 
-    === Linking llsf-refbox ---
-    /usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/4.6/../../../../lib/libmongoclient.a(md5main.o): undefined reference to symbol 'sin@@GLIBC_2.2.5'
-    /usr/bin/ld: note: 'sin@@GLIBC_2.2.5' is defined in DSO /usr/lib/gcc/x86_64-linux-gnu/4.6/../../../x86_64-linux-gnu/libm.so so try adding it to the linker command line
-    /usr/lib/gcc/x86_64-linux-gnu/4.6/../../../x86_64-linux-gnu/libm.so: could not read symbols: Invalid operation
-    collect2: ld returned 1 exit status
+Users attempting to use alternative, unsupported systems are doing so without any guarentees. These are just hints.
+
+1. To install the RoCKIn referee box, please read and install the prerequisites of the LLSF referee box which are described here:
```
Comment (svenschneider) [inline README.md]: 1. "RoCKIn referee box" -> "Central Factory Hub" or "CFH"

File: README.md [hunk 11]
```diff
@@ -43,28 +87,39 @@ How can I solve the problem?
...
+
+3. Older versions of Ubuntu are known to not have all of the above packages. The following may be installed from source if not available:
+    - clips 6.30+ (http://sourceforge.net/projects/clipsmm/files/clips)
+    - clipsmm 0.3.4+(http://sourceforge.net/projects/clipsmm/files/clipsmm)
+    - zeromq 3+ (http://zeromq.org/) 
+    - zeromq.hpp (https://github.com/zeromq/cppzmq/blob/master/zmq.hpp) 
+            
+            cd /usr/local/include/
+            sudo wget https://raw.githubusercontent.com/zeromq/cppzmq/master/zmq.hpp
+    - boost 1.48+ (http://www.boost.org/) It is STRONGLY recommended to install boost through your package manager, and to choose the version which does not conflict with your other dependencies (i.e. ROS)
```
Comment (svenschneider) [inline README.md]: No need to mention ROS.

File: README.md [hunk 12]
```diff
@@ -43,28 +87,39 @@ How can I solve the problem?
...
+  
+4. Cloning the Git repository:
```
Comment (svenschneider) [inline README.md]: Maybe just say to clone and build as described above?

File: README.md [hunk 13]
```diff
@@ -43,28 +87,39 @@ How can I solve the problem?
...
+    
+        git clone https://github.com/rockin-robot-challenge/at_work_central_factory_hub.git
+        cd at_work_central_factory_hub
+        git checkout rockin
```
Comment (svenschneider) [inline README.md]: "rockin" is the default branch, no need to check it out manually.

**Comment by svenschneider** [comment]
Please integrate feedback and update the pull request.

File: README.md (hunk 13 above)
Comment (moriarty) [inline README.md]: That line originally said checkout rockin2015. 
Will remove.

File: README.md (hunk 11 above)
Comment (moriarty) [inline README.md]: Not needed. But I preferred to be clear and explicit.

I'll remove it anyway.

**Comment by moriarty** [comment]
@svenschneider feedback integrated.

**Comment by svenschneider** [comment]
Ok, looks good. Can you please squash the commits appropriately and then update the pull request?

**Comment by moriarty** [comment]
I've squashed the commits and rebased.

**Comment by svenschneider** [comment]
Merged. Thank you.

**Pull request merged**